### PR TITLE
fix(ts): extend `server` type in Email provider from `nodemailer`

### LINKED ISSUE #807: Missing Nodemailer option types
**Summary of proposed feature**
When using typescript we cannot set all node mailer options without resorting to typecasting to any.

**Purpose of proposed feature**
Enable setting other node mailer options like `secure`, `authMethod`, `tls` options, etc.

**Detail about proposed feature**
Either add secure type, or better yet, extend the `ProviderEmailServer` type to extend `SMTPConnection.Options type`:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/623f3889f0fac1ed3284066f990cebf3d86d1d64/types/nodemailer/lib/smtp-connection/index.d.ts#L131

**Potential problems**
N/A

**Describe any alternatives you've considered**
I casted the `server` field to `any`. A suboptimal solution.

**Additional context**
Any other context, screenshots, etc.
image

*Please indicate if you are willing and able to help implement the proposed feature.*
Probably, if time allows.

### PR DESCRIPTION
**Pull request by justinforlenza**
## Reasoning 💡
Adds additional typing for SMTP server config for `tls` & `secure` from nodemailer 

## Checklist 🧢

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

## Affected issues 🎟
Fixes #807

### CODE REVIEW

**Comment by balazsorban44** [comment]
thanks, what about extending the whole option as described in https://github.com/nextauthjs/next-auth/issues/807#issue-731914299?

**Comment by justinforlenza** [comment]
> thanks, what about extending the whole option as described in [#807 (comment)](https://github.com/nextauthjs/next-auth/issues/807#issue-731914299)?

Tbh idk why that just didn't even cross my mind, just made that change to just use the nodemailers built in interface

**Comment by balazsorban44** [comment]
Could you please make sure that you use npm@7 so the lock file doesn't get messed up? 😬

**Comment by justinforlenza** [comment]
> Could you please make sure that you use npm@7 so the lock file doesn't get messed up

Apoligies, I was trying out gitpod, and didnt realize it didnt have updated NPM

**Review by balazsorban44 — APPROVED** [review]

File: types/providers.d.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { Profile, TokenSet, User } from "."
 import { Awaitable, NextApiRequest } from "./internals/utils"
+import SMTPConnection from 'nodemailer/lib/smtp-connection'
```
Comment (balazsorban44) [inline types/providers.d.ts:3]:
```suggestion
import { SMTPConnectionOptions } from 'nodemailer/lib/smtp-connection'
```

File: types/providers.d.ts [hunk 2]
```diff
@@ -149,7 +140,7 @@ export type SendVerificationRequest = (params: {
 export interface EmailConfig extends CommonProviderOptions {
   type: "email"
   // TODO: Make use of https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html
-  server: string | EmailConfigServerOptions
+  server: string | SMTPConnection.Options
```
Comment (balazsorban44) [inline types/providers.d.ts:143]:
```suggestion
  server: string | SMTPConnectionOptions
```

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Our TS tests were still failing. I added some suggestions to mitigate them, and I could seemingly also commit them to your fork, so I did. Hope it's OK. :smile:

File: types/providers.d.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { Profile, TokenSet, User } from "."
 import { Awaitable, NextApiRequest } from "./internals/utils"
+import { SMTPConnectionOptions } from 'nodemailer/lib/smtp-connection'
```
Comment (balazsorban44) [inline types/providers.d.ts:3]:
```suggestion
import { Options as SMTPConnectionOptions } from 'nodemailer/lib/smtp-connection'
```

**Review by balazsorban44 — CHANGES_REQUESTED** [review]

File: types/providers.d.ts [hunk 4]
```diff
@@ -127,16 +128,6 @@ export type CredentialsProvider = (
 
 export type CredentialsProviderType = "Credentials"
```
Comment (balazsorban44) [inline types/providers.d.ts:130]:
```suggestion

```

**Review by balazsorban44 — CHANGES_REQUESTED** [review]

**Review by balazsorban44 — APPROVED** [review]

**Comment by balazsorban44** [comment]
Thank you!

**Pull request merged**